lib/rules/require-aria-activedescendant-tabindex.js

```diff
@@ -0,0 +1,60 @@
+import { dom } from 'aria-query';
+
+import AstNodeInfo from '../helpers/ast-node-info.js';
+import isInteractiveElement from '../helpers/is-interactive-element.js';
+import Rule from './_base.js';
+
+const ERROR_MESSAGE =
+  'A generic element using the aria-activedescendant attribute must have a tabindex';
+
+export default class RequireAriaActivedescendantTabindex extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        let hasAriaActivedescendant = AstNodeInfo.hasAttribute(node, 'aria-activedescendant');
+        if (!hasAriaActivedescendant) {
+          return;
+        }
+        // Bypass validation of Ember components, since we do not know what HTML tags they have
+        if (![...dom.keys()].includes(node.tag)) {
```

Comment:
Possibly a dumb question but how is this checking for Ember components and why does it use aria-query?

Reply:
Since this list from Aria-query contains all known HTML elements, this check assumes any element not included in this list to be a custom component.